starter-code/js/index.js

```diff
@@ -1,11 +1,58 @@
 var $cart = document.querySelector('#cart tbody');
+//console.log($cart);
 var $calc = document.getElementById('calc');
+//console.log($calc);
+
+
+function updateSubtotal(product) {
 
-function updateSubtotal($product) {
   // Iteration 1.1
+  
+  const price = product.querySelector('.pu span').innerHTML;
+  const quantity = product.querySelector('.qty input').value;
+
+  const subTotal = price * quantity;
+
+
+  const subTotalDisplay = product.querySelector('.subtot span');
+  
+  subTotalDisplay.innerText = subTotal;
+  
+  return subTotal;
+  
 }
 
+
 function calculateAll() {
   // Iteration 1.2
+
+  const productArray = document.querySelectorAll('.product');
+  let totalAmount = 0;
+  productArray.forEach(function(product){
+    let subTotal = updateSubtotal(product);
+    totalAmount += subTotal;
+  });
+
+  const grandTotal = document.querySelector('h2 span');
+  grandTotal.innerText = totalAmount;
+
 }
-$calc.onclick = calculateAll;
\ No newline at end of file
+$calc.onclick = calculateAll;
+
+
+//Iteration 2: Add another product (Done in Iteration 1.1)
+//Iteration 3: Calculating the total price of all products (Done in Iteration 1.2)
+
+//Iteration 4: Deleting a product
+
+const deleteButton = document.querySelectorAll('.rm.btn')
```

Comment:
This is an alright start, though you will have to use a bit of .parentNode to be able to delete the specific product. I would personally recommend to loop through the product elements instead of the buttons (like we did in class). Nontheless, what you did here will work in the end:)